lib/puppet/reports/servicenow.rb

```diff
@@ -6,12 +6,18 @@
   include Puppet::Util::Servicenow
 
   def process
-    unless status != 'unchanged' || noop_pending
+    decision_parameters = "status: #{status}, "\
+                          "corrective_change: #{corrective_change}, "\
+                          "noop_pending: #{noop_pending}"
+    Puppet.info("servicenow reporting: decision parameters: #{decision_parameters}")
```

Comment:
Will a user be able to figure out what "decision parameters" means and what they're used for? I'd probably phrase this as `servicenow report processor: using the following parameters to calculate the report attributes: <decision_parameters>`

Reply:
I think `servicenow reporting` => `servicenow report processor` in _all_ the log messages. Latter's more explicit and states where the log message comes from (for some reason Puppet doesn't have a `'processing report via <processor> report processor'` prefix for all the report processors). `servicenow reporting` is vague -- I can definitely imagine a user being confused as to what that term means (e.g. they may wonder what is being reported to ServiceNow, what "reporting to ServiceNow" means [is it via the incident API, some 'reporting' API, ...], etc).

Reply:
Changed to `Puppet.info(sn_log_entry("report attributes parameters: #{decision_parameters}"))`. The function prepends the `report processor` bit and can be extended in the future if we ever want to do fancier stuff with it.